### PR TITLE
Minor refactoring

### PR DESCRIPTION
**Pull request by rsetaluri**
Cleaning up and modernizing some code in preparation for refactoring circuit.py

### CODE REVIEW

File: magma/interface.py [hunk 1]
```diff
@@ -238,27 +214,24 @@ def __str__(cls):
...
 
     def __eq__(cls, rhs):
-        if not isinstance(rhs, InterfaceKind): return False
-
-        if cls.Decl != rhs.Decl: return False
+        if not isinstance(rhs, InterfaceKind):
+            return False
+        if cls.Decl != rhs.Decl:
+            return False
         return True
```
Comment (leonardt) [inline magma/interface.py]: I think we could simplify to just `return cls.Decl == rhs.Decl`

File: magma/port.py [hunk 2]
```diff
@@ -1,39 +1,39 @@
...
         error(message)
+        return
+    file = debug_info[0]
+    line = debug_info[1]
+    message = f"\033[1m{make_relative(file)}:{line}: {message}"
+    fn(message, include_wire_traceback=True)
+    try:
+        fn(get_source_line(file, line))
+    except FileNotFoundError:
+        fn(f"    Could not file file {file}")
```
Comment (leonardt) [inline magma/port.py]: Typo, "file file" should be "find file"

File: magma/ref.py [hunk 3]
```diff
@@ -22,63 +25,67 @@ def qualifiedname(self, sep='.'):
...
         self.name = name
 
-    def qualifiedname(self, sep='.'):
+    def qualifiedname(self, sep="."):
         name = self.name
         if isinstance(self.name, IntegerTypes):
-            # Hack, Hack, Hack
-            if sep == '.':
-                return self.inst.name + '[%d]' % self.name
+            # Hack, Hack, Hack!
```
Comment (leonardt) [inline magma/ref.py]: !!

File: magma/interface.py [hunk 4]
```diff
@@ -82,140 +77,121 @@ def __len__(self):
         return len(self.ports.keys())
 
     def __getitem__(self, key):
+        if isinstance(key, str):
+            return self.ports[key]
+        # TODO(rsetaluri): Is this type of index used? Should we continue to
+        # support it?
         if isinstance(key, int):
-            if isinstance(key,slice):
+            if isinstance(key, slice):
```
[the remark this answers is not in the document]
Comment (rsetaluri) [reply magma/interface.py]: Good catch. Fixed.

File: magma/interface.py [hunk 5]
```diff
@@ -1,74 +1,69 @@
...
+    """
+    if len(decl) % 2:
+        raise ValueError(f"Expected even number of arguments, got {len(decl)}")
+
+    names = decl[::2]
+    ports = decl[1::2]
+    # If name is empty, convert to the index.
+    names = [name if name else str(i) for i, name in enumerate(names)]
+    # Check that all ports are given as instances of Kind or Type.
+    if not all([isinstance(port, (Kind, Type)) for port in ports]):
```
[the remark this answers is not in the document]
Comment (rsetaluri) [reply magma/interface.py]: Done

File: magma/interface.py [hunk 6]
```diff
@@ -82,140 +77,121 @@ def __len__(self):
...
-    #   same as the declaration
     def decl(self):
-        return flatten([name, type(port).flip()] \
-                             for name, port in self.ports.items()  )
-
+        """
+        Return all the arguments as name, flip(port) (same as the declaration).
+        """
+        return _flatten([name, type(port).flip()] \
+                        for name, port in self.ports.items())
```
[the remark this answers is not in the document]
Comment (rsetaluri) [reply magma/interface.py]: Done

File: magma/interface.py [hunk 7]
```diff
@@ -82,140 +77,121 @@ def __len__(self):
...
-#
-# Interface class
-#
-# This function assumes the port instances are provided
-#
-#  e.g. Interface('I0', In(Bit)(), 'I1', In(Bit)(), 'O', Out(Bit)())
-#
+        """Return True if this interface has a Clock."""
+        return any([isinstance(port, ClockType) for \
+                    port in self.ports.values()])
```
[the remark this answers is not in the document]
Comment (rsetaluri) [reply magma/interface.py]: Done

File: magma/interface.py (hunk 1 above)
Comment (rsetaluri) [reply magma/interface.py]: Done

File: magma/port.py (hunk 2 above)
Comment (rsetaluri) [reply magma/port.py]: Done

**Pull request merged**